packages/@aws-cdk/aws-cloudfront/lib/origin_access_identity.ts

```diff
@@ -0,0 +1,126 @@
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/core');
+import { CfnCloudFrontOriginAccessIdentity } from './cloudfront.generated';
+
+export interface CloudFrontOriginAccessIdentityProps {
+  /**
+   * Any comments you want to include about the origin access identity.
+   *
+   * @default "Allows CloudFront to reach to the bucket"
+   */
+  readonly comment?: string;
+}
+
+export interface ICloudFrontOriginAccessIdentity extends cdk.IResource, iam.IGrantable  {
+  /**
+   * The Origin Access Identity Name
+   */
+  readonly name: string;
+}
+
+export abstract class BaseCloudFrontOriginAccessIdentity extends cdk.Resource {
+  /**
+   * The Origin Access Identity Name (physical id)
+   */
+  public abstract readonly name: string;
+  /**
+   * Derived principal value for bucket access
+   */
+  public abstract readonly grantPrincipal: iam.IPrincipal;
+
+  /**
+   * The Arn to include in S3 bucket policy to allow CloudFront access
+   */
+  protected arn(): string {
+    return cdk.Stack.of(this).formatArn(
+      {
+        service: "iam",
+        region: "", // global
+        account: "cloudfront",
+        resource: "user",
+        resourceName: `CloudFront Origin Access Identity ${this.name}`
```

Comment:
what? is that the resource name?

Reply:
yes, it's one of the two ways:

Either by S3 CanonicalUser or via this ARN: https://stackoverflow.com/questions/44877359/aws-where-do-i-see-a-list-of-origin-access-identities